.github/workflows/fetch-dev-artifacts.yaml

```diff
@@ -29,98 +29,96 @@ jobs:
+            aws --no-cli-pager codeartifact get-package-version-asset --domain build-service-live --repository ci-live --format maven --namespace org.neo4j --package cypher-v25-antlr-parser --package-version $neo4j_version_base-SNAPSHOT --asset $grammar_asset grammar.zip --domain-owner ${{ secrets.AWS_ACCOUNT_ID }}
+            
+      - name: fetch semantic analysis artifacts from latest neo4j CI
+        run: | 
+            semantic_asset=$(aws --no-cli-pager codeartifact list-package-version-assets --domain build-service-live --repository ci-live --format maven --namespace com.neo4j --package semantic-analysis-js --package-version $neo4j_version_base-SNAPSHOT --query 'assets[?ends_with(name, `artifacts.zip`)].name | [0]' --output text --domain-owner ${{ secrets.AWS_ACCOUNT_ID }})
+            aws --no-cli-pager codeartifact get-package-version-asset --domain build-service-live --repository ci-live --format maven --namespace com.neo4j --package semantic-analysis-js --package-version $neo4j_version_base-SNAPSHOT --asset $semantic_asset semantic-analysis-js.zip --domain-owner ${{ secrets.AWS_ACCOUNT_ID }}
+      
+      - name: unpack grammar artifacts
+        run: |
+            unzip -j -o grammar.zip \
+              "$(unzip -l grammar.zip | awk '{print $4}' | grep "Cypher25Lexer.g4$")" \
+              "$(unzip -l grammar.zip | awk '{print $4}' | grep "Cypher25Parser.g4$")" \
+              -d ./packages/language-support/src/antlr-grammar
 
-    
-    - name: rename semantic analysis
-      run: mv -f "./packages/language-support/src/syntaxValidation/classes.js" "./packages/language-support/src/syntaxValidation/semanticAnalysis.js"
+      - name: unpack semantic analysis
+        run: |
+          unzip -j -o "semantic-analysis-js.zip" \
+          "$(unzip -l semantic-analysis-js.zip | awk '{print $4}' | grep "classes.js$")" \
+          -d ./packages/language-support/src/syntaxValidation
 
-              
-    - name: clean up zips
-      run: rm -f grammar.zip && rm -f semantic-analysis-js.zip
+      - name: rename semantic analysis
+        run: mv -f "./packages/language-support/src/syntaxValidation/classes.js" "./packages/language-support/src/syntaxValidation/semanticAnalysis.js"
+         
+      - name: clean up zips
+        run: rm -f grammar.zip && rm -f semantic-analysis-js.zip
 
-    - name: Create New Branch Name
-      run: |
-        BRANCH_NAME="update-artifacts-$(date +%Y%m%d-%H%M%S)"
-        echo "BRANCH_NAME=$BRANCH_NAME" >> $GITHUB_ENV
+      - name: Build project
+        uses: ./.github/actions/setup-and-build-project
+      
+      - name: Create New Branch Name
+        run: |
+          BRANCH_NAME="update-artifacts-$(date +%Y%m%d-%H%M%S)"
+          echo "BRANCH_NAME=$BRANCH_NAME" >> $GITHUB_ENV
```

Comment:
GitHub is getting confused but the only changes to this file should have been adding this step to build the project (because we need the new generated TextMate grammar)